cobspec/src/main/java/com/saralein/cobspec/controller/CookieController.java

```diff
@@ -0,0 +1,72 @@
+package com.saralein.cobspec.controller;
+
+import com.saralein.cobspec.data.CookieStore;
+import com.saralein.server.controller.Controller;
+import com.saralein.server.cookies.Cookie;
+import com.saralein.server.cookies.CookieParser;
+import com.saralein.server.request.Request;
+import com.saralein.server.response.Response;
+
+import java.util.List;
+
+public class CookieController implements Controller {
+    private final CookieParser parser;
+    private final CookieStore store;
+
+    public CookieController(CookieParser cookieParser, CookieStore cookieStore) {
+        this.parser = cookieParser;
+        this.store = cookieStore;
+    }
+
+    public Response respond(Request request) {
+        if (hasCookieHeader(request)) {
+            return getCookieResponse(request);
+        }
+
+        return getSetCookieResponse(request);
+    }
+
+    private Response getSetCookieResponse(Request request) {
+        List<Cookie> cookies = parser.parseFromUri(request.getUri());
```

Comment:
Should the `CookieController` be responsible for parsing the cookie? I feel like the parsing should have been completed by the time the request reached to controller.